### DISCUSSION TITLE
brew install soci with mysql backend

### DISCUSSION THREAD

**Discussion by linrongbin16** [Homebrew/discussions#2701] [Getting started] [answered]
Hi, I'm trying to install SOCI (http://soci.sourceforge.net/) with mysql backend.

How should I do ? I didn't find any document about it.

**Comment by SMillerDev** [comment]
You'll have to edit the formula (`brew edit soci`) and then use `brew install --build-from-source soci` to test your changes.

**Comment by linrongbin16** [comment, marked as answer]
<img width="1904" alt="图片" src="https://user-images.githubusercontent.com/6496887/147737706-3a425bd1-eb1b-487c-bdae-f857892a2c1f.png">


Seems soci is hardly usable in macOS/brew.

**Reply by SMillerDev** [reply]
It builds just fine for me on macOS 12:
```console
~ $ brew install -s soci
==> Downloading https://downloads.sourceforge.net/project/soci/soci/soci-4.0.2/soci-4.0.2.zip
Already downloaded: /Users/sean/Library/Caches/Homebrew/downloads/19ca3e71256cdce3a73d50209d462ace9d45351dd37cbe7e04da03fd4bc7f4eb--soci-4.0.2.zip
==> cmake .. -DSOCI_TESTS:BOOL=OFF -DWITH_SQLITE3:BOOL=ON -DWITH_BOOST:BOOL=OFF -DWITH_MYSQL:BOOL=OFF -DWITH_ODBC:BOOL=OFF -DWITH_ORACLE:BOOL=OFF -DWITH_POSTGRESQL:BOOL=OFF
==> make install
Warning: Libraries were compiled with a flat namespace.
This can cause linker errors due to name collisions, and
is often due to a bug in detecting the macOS version.
  /usr/local/Cellar/soci/4.0.2/lib/libsoci_empty.4.0.2.dylib
  /usr/local/Cellar/soci/4.0.2/lib/libsoci_sqlite3.4.0.2.dylib
==> Summary
🍺  /usr/local/Cellar/soci/4.0.2: 71 files, 1.5MB, built in 29 seconds
==> Running `brew cleanup soci`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

I suspect the change you made broke it.

**Reply by linrongbin16** [reply]
Do I need to install `brew install mysql-client` as a dependency first?

**Reply by SMillerDev** [reply]
I'm not sure what you need to change. I didn't write the software, nor do I use it.

Just installing software won't edit the build environment though, you will need to add a `depends_on "mysql-client"` for brew to expose it to the soci build.
